File: extensions/wi/example/sources.cpp
#include <iostream>
#include <string>
#include <memory>

namespace math {
  // Example asyncAdd function
  int asyncAdd(int a, int b) { return a + b; }
} // namespace math

class Container {
  // Base class for container functionality (empty for now)
};

class A {
  // Base class A (empty for now)
};

class B {
  // Base class B (empty for now)
};

class Player : public Container, public A, public B {
public:
  int32_t x = 3;

  int32_t add(int32_t a, int32_t b, int32_t c = 5) { return a + b + c; }

  void update() {
    int8_t y = 255;
    float z = 100;

    // Simulate asyncAdd functionality
    int response = math::asyncAdd(1, 2);

    if (!response) {
      throw std::runtime_error("Error");
    }

    x = x * (y - z) / -(100 + x);
  }

private:
  bool canMove = false;
  std::string str = "LOONG STRING";

  std::string getName() { return "NeutrinoZh"; }
};

class Board : public Container, public A, public B {
public:
  void example(bool a) {
    for (int i = 0; i < 10; ++i) {
      i += 1;
    }

    bool i = false;
    while (!i || (true && a)) {
      i = true;
      if (false) {
        break;
      } else {
        continue;
      }
    }

    if (false) {
      bool b = true;
      bool c = true;
      return;
    }

    if (true) {
      return;
    }
  }
};

int32_t add(int32_t a, int32_t b, int32_t c = 5) { return 132; }

int main() {
  std::unique_ptr<Player> c = std::make_unique<Player>();
  std::unique_ptr<Board> s = std::make_unique<Board>();

  return 0;
}

int32_t outerFunction(int32_t a, int32_t b, int32_t c = 5) { return 132; }

File: extensions/wi/example/sources.js
import * as math from 'math';

const re = /ab+c/i;

export class Player {
  constructor() {
    this.x = 3;
  }

  add(a, b, c = 5) {
    return a + b + c;
  }

  async update() {
    let y = 255;
    let z = 100;

    // Simulate asyncAdd functionality
    let response = await math.asyncAdd(1, 2);

    if (!response) {
      // throw new Error('Error');
      throw new Error(`${y}`);
    }

    this.x = (this.x * (y - z)) / -(100 + this.x);
  }

  getName() {
    return 'NeutrinoZh';
  }

  // private field simulation
  _canMove = false;
  _string = 'LOONG STRING';
}

export class Board {
  async example(a) {
    let i = 0;
    for (i = 0; i < 10; ++i) {
      i += 1;
    }

    while (false || (true && a)) {
      i = 10;
      if (false) {
      } else {
        continue;
      }
    }

    if (true) {
      return true;
    }
  }
}

export function add(a, b, c = 5) {
  return 132;
}

// Create instances of classes
const c = new Player();
const s = new Board();

export async function outerFunction(a, b, c = 5) {
  return 132;
}
